src/helpers.js

```diff
@@ -95,10 +96,13 @@ function getProperPayload (event) {
   let decodedJsonPayload
   try {
     if (isBase64Encoded(event)) {
-      decodedJsonPayload = JSON.parse(Buffer.from(event, 'base64').toString('utf-8'))
+      // using JSONbig.parse to handle payloads which may contain big integers.
+      // cf https://www.irt.org/script/1031.htm
+      // using JSON.parse, there will be data loss as these big numbers outside the range gets transformed to nearest Javascript number
+      decodedJsonPayload = JSONbig.parse(Buffer.from(event, 'base64').toString('utf-8'))
     } else {
       // parsing for non-encoded json payloads (e.g. custom events)
-      decodedJsonPayload = JSON.parse(event)
+      decodedJsonPayload = JSONbig.parse(event)
```

Comment:
Does this guarantee ordering of keys because even that could cause the signature verification to fail? 
I would suggest working with the raw payload instead (without any JSON parsing). WDYT?

Reply:
It's built on top of the `JSON.parse` and I tested multiple times which parses payload just fine. Working with raw payload wouldn't work as we need to fetch the `recipient_client_id` from the payload, which I am not sure if we can get unless we parse the json we receive

Reply:
We can verify the signature first, then verify the `recipient_client_id` by parsing the json payload so that the order of keys does not matter anymore. WDYT?

Reply:
what we can do is use the parsed payload to verify the `recipient_client_id` and then use the raw payload (which I beleive won't change at this point here) for the signature verification. Else the whole verification sequence has to be changed in the code. WDYT?

Reply:
Okay. I will verify the above and change accordingly.